### PR TITLE
Starting out network bridge package

### PR DESCRIPTION
**Pull request by GASB0**

### CODE REVIEW

File: configs/cm3_defconfig [hunk 1]
```diff
@@ -23,6 +23,9 @@ BR2_PACKAGE_HOST_MTOOLS=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS=y
 BR2_PACKAGE_HOST_GENIMAGE=y
 
+# Misc packages
```
Comment (flatmax) [inline configs/cm3_defconfig]: change this to someting like : # bridging eth to usb

File: package/usb-bridge/Config.in [hunk 2]
```diff
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_USB_BRIDGE
+  bool "usb-bridge"
+  select USB_ETH
```
Comment (flatmax) [inline package/usb-bridge/Config.in]: Get rid of this select - I don't think it is necessary

File: package/usb-bridge/scripts/S41-start-bridge [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+bridgeName="br0"
+etherInterface="eth0"
+
+if [ "$1" == "start" ]; then
+  # Loading g_ether and creating the appropriate network interface
+  modprobe dwc3-of-simple
+  modprobe dwc3-rockchip-inno
+
+  modprobe g_ether dev_addr=12:34:56:78:9a:bc host_addr=12:34:56:78:9a:bd
+  ip address add 192.0.2.1/30 dev usb0
```
Comment (flatmax) [inline package/usb-bridge/scripts/S41-start-bridge]: get rid of static mac and addresses. Your router will allocate IP addresses. You see, if you have more then one device on the network they can't have the same MACs and IP.

File: package/usb-bridge/scripts/S41-start-bridge [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+bridgeName="br0"
+etherInterface="eth0"
+
+if [ "$1" == "start" ]; then
```
Comment (flatmax) [inline package/usb-bridge/scripts/S41-start-bridge]: change if statements to a case statemement.

File: package/usb-bridge/usb-bridge.mk [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+USB_BRIDGE_VERSION = 0.0.1
+USB_BRIDGE_LICENSE = GPL-3.0+
+USB_BRIDGE_SITE = /home/gabriel/Documents/Projects/buildroot.rockchip.ext/package/usb-bridge/scripts
+USB_BRIDGE_SITE_METHOD = local
+USB_BRIDGE_STAGING=YES
+
+define USB_BRIDGE_INSTALL_TARGET_CMDS
+	cp $(@D)/S41-start-bridge $(TARGET_DIR)/etc/init.d/
+	chmod 555 $(TARGET_DIR)/etc/init.d/S41-start-bridge
+endef
```
Comment (flatmax) [inline package/usb-bridge/usb-bridge.mk:16]: great ! does it install and run at boot time ?

**Review by flatmax — CHANGES_REQUESTED** [review]
Test this setup by linking your laptop to the internet through usb networking. Turn off your laptop's wifi to make that happen.

File: package/usb-bridge/usb-bridge.mk (hunk 5 above)
Comment (GASB0) [reply package/usb-bridge/usb-bridge.mk:16]: I'll have to check this one when the power comes back.

File: configs/cm3_defconfig [hunk 6]
```diff
@@ -23,6 +23,10 @@ BR2_PACKAGE_HOST_MTOOLS=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS=y
 BR2_PACKAGE_HOST_GENIMAGE=y
 
+# Bridging Eth to USB
+BR2_PACKAGE_USB_BRIDGE=y
+BR2_SYSTEM_DHCP="br0"
```
Comment (flatmax) [inline configs/cm3_defconfig:28]: below in the the same file you declared eth0 as the BR2_SYSTEM_DHCP. Remove the one for eth0.

File: package/usb-bridge/scripts/S41-start-bridge [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    # Removing the kernel modules
+    rmmod g_ether
+    rmmod dwc3-of-simple
+    rmmod dwc3-rockchip-inno
+
+    # Getting the eth0 interface back to normal
+    ip link set dev $etherInterface down
+    ip address add 0.0.0.0 dev $etherInterface
+    ip link set dev $etherInterface up
+    udhcpc -i $etherInterface
```
Comment (flatmax) [inline package/usb-bridge/scripts/S41-start-bridge:48]: Great work !

**Review by flatmax — CHANGES_REQUESTED** [review]

**Pull request merged**